File: src/cli/team/states.ts
import type { Command } from "commander";
import { getClient } from "../../lib/client.ts";
import { printError, printJson } from "../../lib/output.ts";

export function registerStates(team: Command): void {
  team
    .command("states")
    .description("List workflow states for a team (discover valid status values)")
    .argument("<team>", "Team key or name")
    .action(async (teamInput: string) => {
      try {
        const client = getClient();
        const states = await client.workflowStates({
          filter: {
            team: {
              or: [{ key: { eqIgnoreCase: teamInput } }, { name: { eqIgnoreCase: teamInput } }],
            },
          },
        });
        if (states.nodes.length === 0) {
          printError(`No workflow states found for team "${teamInput}".`);
          return;
        }
        const sorted = [...states.nodes].sort((a, b) => a.position - b.position);
        printJson(
          sorted.map((s) => ({
            id: s.id,
            name: s.name,
            type: s.type,
            color: s.color,
            position: s.position,
          })),
        );
      } catch (err) {
        printError(err instanceof Error ? err.message : "Failed to list workflow states");
      }
    });
}

File: internal/cli/team/states.go
package team

import (
	"context"
	"fmt"
	"sort"

	"github.com/spf13/cobra"

	"github.com/shhac/lin/internal/linear"
	"github.com/shhac/lin/internal/output"
	"github.com/shhac/lin/internal/ptr"
	"github.com/shhac/lin/internal/resolvers"
)

func registerStates(team *cobra.Command) {
	cmd := &cobra.Command{
		Use:   "states <team>",
		Short: "List workflow states for a team (discover valid status values)",
		Args:  cobra.ExactArgs(1),
		Run: func(_ *cobra.Command, args []string) {
			teamInput := args[0]
			client := linear.GetClient()

			team, err := resolvers.ResolveTeam(client, teamInput)
			if err != nil {
				output.PrintError(err.Error())
			}

			filter := &linear.WorkflowStateFilter{
				Team: &linear.TeamFilter{Id: &linear.IDComparator{Eq: ptr.To(team.ID)}},
			}
			resp, err := linear.WorkflowStates(context.Background(), client, filter)
			if err != nil {
				output.PrintError(err.Error())
			}

			nodes := resp.WorkflowStates.Nodes
			if len(nodes) == 0 {
				output.PrintError(fmt.Sprintf("No workflow states found for team %q.", teamInput))
			}

			sort.Slice(nodes, func(i, j int) bool {
				return nodes[i].Position < nodes[j].Position
			})

			items := make([]map[string]any, len(nodes))
			for i, s := range nodes {
				items[i] = map[string]any{
					"id":       s.Id,
					"name":     s.Name,
					"type":     s.Type,
					"color":    s.Color,
					"position": s.Position,
				}
			}

			output.PrintJSON(items)
		},
	}
	team.AddCommand(cmd)
}
